src/main/java/com/jcraft/jsch/ChannelSftp.java

```diff
@@ -516,7 +520,13 @@ public void _put(InputStream src, String dst, SftpProgressMonitor monitor, int m
           // System.err.println(eee);
         }
       }
-      if (mode == RESUME && skip > 0) {
+    if((mode==RESUME&& chunkStart!=null) && skip>0){
+  	    long skipped=src.skip(skip-Long.parseLong(chunkStart));
+  	    if(skipped<(skip-Long.parseLong(chunkStart))){
+  	    	throw new SftpException(SSH_FX_FAILURE, "failed to resume for "+dst);
+  	    }
+  	 }
+  	else if(mode==RESUME && skip>0){
```

Comment:
I think this could be better written as simply something like this to avoid code duplication:
```
      if (mode == RESUME && skip > 0) {
        long srcSkip = chunkStart != null ? skip - Long.parseLong(chunkStart) : skip;
        long skipped = src.skip(srcSkip);
        if (skipped < srcSkip) {
          throw new SftpException(SSH_FX_FAILURE, "failed to resume for " + dst);
        }
      }
```

Reply:
I have made the changes as suggested.